src/BuiltInTools/dotnet-watch/BrowserRefreshServer.cs

```diff
@@ -38,11 +38,14 @@ public async ValueTask<string> StartAsync(CancellationToken cancellationToken)
         {
             var hostName = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1";
 
+            hostName = hostName.StartsWith("ws") ? hostName.Replace("ws","http") : $"http://{hostName}";
```

Comment:
Could we do something like this? Also this would warrant a few unit tests. 

```suggestion
            var configuredEnv = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME");
            var url = GetAutoReloadUrl(configuredEnv);
            
            internal static string GetAutoReloadUrl(string configuredEnv)
            {
                var url = "http://127.0.0.1:80";
                if (configuredEnv is null)
                {
                   return url;
                }
                
                if (Uri.TryCreate(configuredEnv , UriKind.Absolute, out url))
                {
                   return url;
                }
                
                return $"http://{hostName}:0";
           }
```

Reply:
Yes, I will update into branch.

Reply:
@CliXiD @pranavkm Just curiosity, I wonder if  `DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME` can be both absolute URI and host name can confuse people who use this option or we need to add more document on how to use this option as well.

I am thinking if we can introduce a new environment variable `DOTNET_WATCH_AUTO_RELOAD_WS_PORT` to be similar to Jekyll which has [--livereload-port option](https://jekyllrb.com/docs/configuration/options/#serve-command-options)

Then we can change only 2 lines of code.
```c#
var hostName = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1"; // existing behavior

var port = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_PORT") ?? "0"; // 1st change, new env var

            _refreshServer = new HostBuilder()
                .ConfigureWebHost(builder =>
                {
                    builder.UseKestrel();
                    builder.UseUrls($"http://{hostName}:{port}");  // 2nd change where we specify port number

                    builder.Configure(app =>
                    {
                        app.UseWebSockets();
                        app.Run(WebSocketRequest);
                    });
                })
                .Build();
```

Reply:
I don't want to introduce new environment since it will cause more work for the CI tools of this SDK.

Reply:
@CliXiD could you please explain in details why having an optional `DOTNET_WATCH_AUTO_RELOAD_WS_PORT` make more work for the dotnet CLI tool?

Reply:
Just in case if there is test cases for BrowserRefreshServer in the CI and also more work on documentation.
I don't think the documentation will be difficult to understand. We can just simply add example of accepted values.

Also, if I'm correct, we can't let BrowserRefreshServer listen on 127.0.0.1 in container because the container's ip is not 127.0.0.1 so the connection will not accept. 
So, to let development live reload work. You have to assign DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME with ws://[::]:1234 or ws://*:1234

Reply:
CliXiD Thanks you for explaining me. 
My idea is that it would be nice if we can publish a live reload port from a container to a host like what I did in this Docker Compose https://github.com/codesanook/Codesanook.Examples/blob/master/codesanook-examples-jekyll/docker-compose.yml#L14

I don't know how Jekyll live reload  work under the hood.
However, since it uses a static live reload port, I can publish a container's port to the a host.

If you are interested in Jekyll project, here is how to run it locally.
https://github.com/codesanook/Codesanook.Examples/tree/master/codesanook-examples-jekyll#to-run-the-project-locally

I have tested Jekyll hot reload with some endpoints, here are the results.
- Testing with http://localhost:4000 work both website and hot reload.
- Testing with http://127.0.0.1:4000 work both website and hot reload.
- Testing with http://0.0.0.0:4000 does not work
- Environment, code checked out and run in WSL2 on Windows 10 pro version 20H2, build 19042.928

Reply:
FYI, how Jekyll hot reload works in both normal local development and with a Docker container.
- For normal local development (without Docker container):
    - Use a default host name from https://github.com/jekyll/jekyll/blob/master/lib/jekyll/configuration.rb#L48

- For a Docker container:
    - It override "host" value with [-H host parameter](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/commands/serve.rb#L17)
    by [Docker image argument](https://github.com/envygeeks/jekyll-docker/blob/master/repos/jekyll/copy/all/usr/jekyll/bin/default-args#L22).

Now it is clear to me that CliXiD has suggested that we need to specific both DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME and DOTNET_WATCH_AUTO_RELOAD_WS_PORT values
if add DOTNET_WATCH_AUTO_RELOAD_WS_PORT variable.

For example:
- Set DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME to [::] or * or 0.0.0.0 as Jekyll Docker image does. https://github.com/envygeeks/jekyll-docker/blob/master/repos/jekyll/copy/all/usr/jekyll/bin/default-args#L22
- Set DOTNET_WATCH_AUTO_RELOAD_WS_PORT to a specific number.

Reply:
You got my answer from private chat already. So in summary I dont want to intrduce a new environment.

To answer if it possible to have a new environment and configure only port.

The answer is yes, we can check if there is env DOTNET_USE_POLLING_FILE_WATCHER that declared in https://github.com/dotnet/dotnet-docker/blob/main/src/sdk/5.0/buster-slim/amd64/Dockerfile#L9

if it's declared, the default host will be ws://[::] and you can set only port or with some default fix port eg. 5002
if it's not declare, the host name logic will remaining the same

@pranavkm  if you are okay to introduce the new environment, what do you think for the below approach?

```csharp
var isRunningInDocker  = Environment.GetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER") != null;
var configuredHostname = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") || "127.0.0.1";
var configuredPort = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_PORT") || "5002";

var url = isRunningInDocker ? $"http://[::]:{configuredPort}" : $"http://{configuredHostname}:0";
```

Reply:
@CliXiD Thanks for following up.